src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/NetHttpTestServiceHost.cs

```diff
@@ -32,4 +32,30 @@ public NetHttpTestServiceHost(Type serviceType, params Uri[] baseAddresses)
         {
         }
     }
+
+    public class NetHttpTestServiceHostUsingWebSocketsFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            NetHttpTestServiceHostUsingWebSockets serviceHost = new NetHttpTestServiceHostUsingWebSockets(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class NetHttpTestServiceHostUsingWebSockets : TestServiceHostBase<IWcfService>
+    {
+        protected override string Address { get { return "NetHttpWebSockets"; } }
+
+        protected override Binding GetBinding()
+        {
+            var binding = new NetHttpBinding();
+            binding.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
```

Comment:
Because the test uses IDuplexSessionChannel the incoming message will be using WebSockets, and because the interface on the server side is not attributed with "CallbackContract" the binding must explicitly set the WebSocketTransportUsage to Always, otherwise the call fails.